### PR TITLE
Firewall improvements

### PR DESCRIPTION
**Pull request by andrewsomething**
This PR includes a number of improvements to the Firewall support recently added in #220. It adds a few new data structure to hold outbound and inbound rules for Firewalls, adds a create method, adds docstrings, and adds an example to the README.

CC : @chrisella

### CODE REVIEW

**Comment by koalalorenzo** [comment]
Can you fix the problems reported in the codebeat report?

**Comment by andrewsomething** [comment]
I fixed the warning level issues around nesting level, but not quite sure how to go about handling the info level Assignment Branch Condition stuff. There's just a lot to assign! :smile:

File: digitalocean/Firewall.py [hunk 1]
```diff
@@ -1,16 +1,142 @@
...
+
+
+class _targets(object):
+    """
+    An internal object that both `Sources` and `Destinations` derive from.
+
+    Not for direct use by end users.
+    """
+    def __init__(self, addresses=[], droplet_ids=[],
+                 load_balancer_uids=[], tags=[]):
```
Comment (koalalorenzo) [inline digitalocean/Firewall.py]: Too many arguments according to code beat. Meh 
Just make him happy, plz! 🤣

**Comment by koalalorenzo** [comment]
Any update on this?

**Pull request merged**